<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title> • EpiForBioWorkshop2020</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="">
<meta property="og:description" content="EpiForBioWorkshop2020">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EpiForBioWorkshop2020</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/EpiForBioWorkshop.html">UNKNOWN TITLE</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/seandavi/BuildABiocWorkshop2020">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="EpiForBioWorkshop_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip></h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/cmirzayi/EpiForBioWorkshop2020/blob/master/vignettes/EpiForBioWorkshop.Rmd"><code>vignettes/EpiForBioWorkshop.Rmd</code></a></small>
      <div class="hidden name"><code>EpiForBioWorkshop.Rmd</code></div>

    </div>

    
    
<div id="epidemiology-for-bioinformaticians" class="section level1">
<h1 class="hasAnchor">
<a href="#epidemiology-for-bioinformaticians" class="anchor"></a>Epidemiology for Bioinformaticians</h1>
<div id="instructors-names-and-contact-information" class="section level2">
<h2 class="hasAnchor">
<a href="#instructors-names-and-contact-information" class="anchor"></a>Instructor(s) name(s) and contact information</h2>
<p>Chloe Mirzayi, <a href="mailto:cmirzayi@gmail.com" class="email">cmirzayi@gmail.com</a></p>
<p>Levi Waldron, <a href="mailto:levi.waldron@sph.cuny.edu" class="email">levi.waldron@sph.cuny.edu</a></p>
<p>CUNY School of Public Health, 55 W 125th St, New York, NY 111027</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">curatedMetagenomicData</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">dagitty</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">DESeq2</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">phyloseq</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggdag</span>)</pre></body></html></div>
</div>
<div id="introduction-just-what-is-a-cause-anyway" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction-just-what-is-a-cause-anyway" class="anchor"></a>Introduction: Just what is a cause anyway?</h2>
<p>Dictionary definition</p>
<blockquote>
<p>something that brings about an effect or a result (Merriam-Webster)</p>
</blockquote>
<p>This doesn’t tell us much though in terms of identifying a cause. How do we know our potential cause brought about an effect or a result? Perhaps it preceded the effect of interest. However, it could still just be a coincidence. Perhaps there is a lurking second cause that causes both our potential cause and the observed effect.</p>
<p>Despite this, humans (and animals) tend to have a relatively intuitive understanding of causation when we observe it directly. When I burn my finger while cooking, I know what caused it–my finger coming into contact with a hot stove. I could also trace the causal pathway backward and identify what caused my finger to come into contact with the hot stove in the first place–such as my motivation to cook dinner or me being distracted by the antics of my cat.</p>
<p><strong>Example:</strong> My cat, who I adopted in 2017, has quickly learned that jumping on my kitchen table while I’m eating results in him getting a spritz of water with a spray bottle–but only if the spray bottle is within reach. <strong>What potential causal variables could my cat be considering when he decides to jump on the table?</strong></p>
</div>
<div id="the-counterfactual-definition-of-a-cause" class="section level2">
<h2 class="hasAnchor">
<a href="#the-counterfactual-definition-of-a-cause" class="anchor"></a>The counterfactual definition of a cause</h2>
<p>However, the nature of much of the work I (and probably most of you) do does not allow us to directly observe what caused an event of interest. Instead we must rely on inferences we draw from our data to establish an argument for a particular causal mechanism or pathway.</p>
<p>Rather than relying on vague dictionary definitions or me talking at length about my cat to inform the science of causal inference, the most commonly accepted definition of a cause in modern epidemiology is based on a counterfactual: <em>what would have happened had the event of interest had the exposure not occurred</em>.</p>
<p>But that’s not very intuitive or easy to parse so let’s consider an example:</p>
<div id="example-a-dog-and-an-ambulance" class="section level3">
<h3 class="hasAnchor">
<a href="#example-a-dog-and-an-ambulance" class="anchor"></a>Example: A dog and an ambulance</h3>
<p>An ambulance drives by a house with its sirens on. A dog in the house barks. We can ask the causal question in a straightforward way: <strong>Did the ambulance driving by the house cause the dog to bark?</strong></p>
<p>Alternatively, we can rephrase the question in terms of a counterfactual: Would the dog have barked if the ambulance had not driven by the house? We can visualize the relationship using a causal diagram also called a directed acyclic graph (DAG).</p>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-2-1.png" width="288"></p>
<p>This diagram depicts the cause or exposure (the ambulance) and the effect or the outcome (the dog barking) with an arrow indicating the direction of the causal relationship. In contrast to a diagram showing a statistical relationship, causal diagrams must state a directional relationship because we are theorizing that one variable causes another.</p>
<p>This is the simplest form of a causal diagram. If we were to statistically model the effect for this relationship using regression, a simple two-variable model containing the exposure and the outcome would give us a correct, unbiased estimate of the actual strength of the cause on the effect. However, we are rarely that lucky. A common source of bias is confounding.</p>
</div>
</div>
<div id="bias-and-confounding" class="section level2">
<h2 class="hasAnchor">
<a href="#bias-and-confounding" class="anchor"></a>Bias and Confounding</h2>
<div id="confounding" class="section level3">
<h3 class="hasAnchor">
<a href="#confounding" class="anchor"></a>Confounding</h3>
<p>Confounding arises when a third variable is present in the relationship between cause-and-effect. A confounder is present when:</p>
<ol style="list-style-type: decimal">
<li>The confounder causes the outcome</li>
<li>The confounder causes the exposure</li>
<li>The confounder is not a mediator (i.e. it is not present on the causal pathway between the exposure and the outcome)</li>
</ol>
<p>Returning to our example, the relationship between the ambulance and the barking dog could be confounded by a third variable: a nearby car crash. Perhaps this car crash is what started the dog barking before the ambulance even arrived, but the ambulance caused the dog to bark even more.</p>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-3-1.png" width="288"> The pathway between the exposure and the outcome through the confounder is called a “backdoor pathway” and it can bias the relationship between the exposure and the outcome you observe. In this situation, we might observe barking and conclude that it occurred due to the ambulance when in fact it was due to the car crash. Or we might conclude that the volume of the barking caused by the ambulance alone is much greater than it actually is due to the combined effect of both the car crash and the ambulance.</p>
<p>We can “deconfound” the effect of interest by controlling for the confounder. The old-fashioned way of doing this is to manually stratify your data by levels of the confounder. Then you calculate an effect size for each stratum. In a modern regression model, we can include the confounder as an independent variable in our model and it is controlled for. When we include the confounder as a variable in a regression model, we are effectively stratifying the effect across different levels of the confounder then taking the weighted average across strata to get an effect size.</p>
</div>
<div id="colliders" class="section level3">
<h3 class="hasAnchor">
<a href="#colliders" class="anchor"></a>Colliders</h3>
<p>Taken at face value, colliders are similar to confounders. However, colliders are not common causes of the exposure and outcome. Instead:</p>
<ol style="list-style-type: decimal">
<li>The outcome causes the collider</li>
<li>The exposure causes the collider</li>
<li>The collider is not a mediator</li>
</ol>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-4-1.png" width="288"></p>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-5-1.png" width="768"></p>
<p>Adjusting for a collider as one would for a confounder can create bias. In effect by adjusting for a collider, a backdoor pathway is opened between the exposure and the outcome through the collider. So what’s the proper way of dealing with colliders? Ignoring them! Hypothesized colliders should not be adjusted for or included in models.</p>
<div id="side-note-on-pathways" class="section level4">
<h4 class="hasAnchor">
<a href="#side-note-on-pathways" class="anchor"></a>Side note on pathways</h4>
<p>In a DAG, a pathway is considered blocked: - If there are two arrows pointing into each other along the path (as in the case of a collider) - If you have deconfounded (controlled, stratified, adjusted) for a variable on the pathway</p>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-6-1.png" width="576"></p>
</div>
</div>
<div id="selection-bias" class="section level3">
<h3 class="hasAnchor">
<a href="#selection-bias" class="anchor"></a>Selection bias</h3>
<p>Another major issue of causal inference is selection bias. This arises when selection is dependent on the exposure and the outcome in the study:</p>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-7-1.png" width="288"></p>
<p>As can be seen from the DAG above, selection bias is a collider. In this case, the study is conditioned on the collider by the act of selection for the study. As a result, the study is biased. One particularly memorable way to think of selection bias is as the “already dead” problem. People who have already died of the outcome are not alive to be in the study. These people may have a more aggressive or serious form of the outcome and not including them masks some of the causal relationship.</p>
<p>A common source of selection bias is loss to follow-up in many studies. In survival analysis in particular this can be important as the people who are lost may differ in important ways from those who completed the study. Unfortunately because these people were lost, it is often difficult to assess how they differed from those who were not lost.</p>
</div>
</div>
<div id="study-designs" class="section level2">
<h2 class="hasAnchor">
<a href="#study-designs" class="anchor"></a>Study Designs</h2>
<div id="randomized-control-trials-rcts" class="section level3">
<h3 class="hasAnchor">
<a href="#randomized-control-trials-rcts" class="anchor"></a>Randomized Control Trials (RCTs)</h3>
<ul>
<li>Exposure of interest is assigned at random</li>
<li>Because exposure is assigned at random, there are no possible confounders of the relationship between exposure and outcome</li>
<li>Considered the gold standard for causal inference</li>
<li>Selection bias is still possible</li>
</ul>
<p><strong>Question:</strong> How can selection bias occur in an RCT?</p>
<div id="instrumental-variable-analysismendelian-randomization" class="section level4">
<h4 class="hasAnchor">
<a href="#instrumental-variable-analysismendelian-randomization" class="anchor"></a>Instrumental Variable Analysis/Mendelian Randomization</h4>
<ul>
<li>Attempts to mimic an RCT using a source of pseudorandomization</li>
</ul>
<p>Consider the DAG for an RCT:</p>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-8-1.png" width="480"></p>
<p>In IV/MR analysis the coin flip is replaced with a source of randomization such as genetic variation: <img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-9-1.png" width="480"></p>
<p>Researchers have used a variety of factors as sources of pseudorandomness in IV studies: - Before and after implementation of a new government policy - Distance from a medical center -</p>
</div>
</div>
<div id="cohort-studies" class="section level3">
<h3 class="hasAnchor">
<a href="#cohort-studies" class="anchor"></a>Cohort Studies</h3>
<ul>
<li>Participants are chosen on a common characteristic (such as occupation, location, or history of a particular disease)</li>
<li>They are then followed over time and exposures and outcomes can be assessed</li>
<li>Selection bias and confounding can occur</li>
</ul>
</div>
<div id="case-control-studies" class="section level3">
<h3 class="hasAnchor">
<a href="#case-control-studies" class="anchor"></a>Case-Control Studies</h3>
<ul>
<li>Participants are chosen based on having an outcome of interest (cases)</li>
<li>Then controls are selected from a similar population to the cases (often matched on age, sex, and other demographic variables)</li>
<li>Selection bias and confounding can occur</li>
</ul>
</div>
<div id="genome-wide-association-study-gwas" class="section level3">
<h3 class="hasAnchor">
<a href="#genome-wide-association-study-gwas" class="anchor"></a>Genome-Wide Association Study (GWAS)</h3>
<ul>
<li><p>Observational study of phenotypes</p></li>
<li><p>Often participants are chosen based on a specific phenotype or disease</p></li>
<li><p>Can be a case-control or a cohort study</p></li>
</ul>
<p><strong>Question</strong>: What are some possible sources of confounding or bias in a GWAS?</p>
</div>
</div>
<div id="causal-inference-in-r" class="section level2">
<h2 class="hasAnchor">
<a href="#causal-inference-in-r" class="anchor"></a>Causal Inference in R</h2>
<p>R provides many packages that are helpful in causal inference.</p>
<div id="dagitty" class="section level3">
<h3 class="hasAnchor">
<a href="#dagitty" class="anchor"></a>Dagitty</h3>
<p>Dagitty allows for creating causal diagrams, but also gives you more information including what variables to adjust for in complicated causal models.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="no">g1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dagitty/man/dagitty.html">dagitty</a></span>( <span class="st">"dag {
    W1 -&gt; Z1 -&gt; X -&gt; Y
    Z1 &lt;- V -&gt; Z2
    W2 -&gt; Z2 -&gt; Y
    X &lt;- W1 -&gt; W2 -&gt; Y
}"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/dagitty/man/graphLayout.html">graphLayout</a></span>(<span class="no">g1</span>))</pre></body></html></div>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-10-1.png" width="576"></p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="kw">for</span>( <span class="no">n</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">g1</span>) ){
    <span class="kw">for</span>( <span class="no">m</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">setdiff</a></span>( <span class="fu"><a href="https://rdrr.io/pkg/dagitty/man/AncestralRelations.html">descendants</a></span>( <span class="no">g1</span>, <span class="no">n</span> ), <span class="no">n</span> ) ){
        <span class="no">a</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dagitty/man/adjustmentSets.html">adjustmentSets</a></span>( <span class="no">g1</span>, <span class="no">n</span>, <span class="no">m</span> )
        <span class="kw">if</span>( <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">a</span>) <span class="kw">&gt;</span> <span class="fl">0</span> ){
            <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span>(<span class="st">"The total effect of "</span>,<span class="no">n</span>,<span class="st">" on "</span>,<span class="no">m</span>,
                <span class="st">" is identifiable controlling for:\n"</span>,<span class="kw">sep</span><span class="kw">=</span><span class="st">""</span>)
            <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>( <span class="no">a</span>, <span class="kw">prefix</span><span class="kw">=</span><span class="st">" * "</span> )
        }
    }
}</pre></body></html></div>
<pre><code>## The total effect of V on Z2 is identifiable controlling for:
##  *  {}
## The total effect of V on Y is identifiable controlling for:
##  *  {}
## The total effect of V on Z1 is identifiable controlling for:
##  *  {}
## The total effect of V on X is identifiable controlling for:
##  *  {}
## The total effect of W1 on Z1 is identifiable controlling for:
##  *  {}
## The total effect of W1 on X is identifiable controlling for:
##  *  {}
## The total effect of W1 on Y is identifiable controlling for:
##  *  {}
## The total effect of W1 on W2 is identifiable controlling for:
##  *  {}
## The total effect of W1 on Z2 is identifiable controlling for:
##  *  {}
## The total effect of W2 on Z2 is identifiable controlling for:
##  *  {}
## The total effect of W2 on Y is identifiable controlling for:
##  *  { V, X }
##  *  { W1 }
## The total effect of X on Y is identifiable controlling for:
##  *  { W2, Z2 }
##  *  { V, W2 }
##  *  { V, W1 }
##  *  { W1, Z1 }
## The total effect of Z1 on X is identifiable controlling for:
##  *  { W1 }
## The total effect of Z1 on Y is identifiable controlling for:
##  *  { W1, W2, Z2 }
##  *  { V, W1 }
## The total effect of Z2 on Y is identifiable controlling for:
##  *  { W2, X }
##  *  { W1, W2, Z1 }
##  *  { V, W2 }</code></pre>
</div>
</div>
<div id="example-in-curatedmetagenomicdata" class="section level2">
<h2 class="hasAnchor">
<a href="#example-in-curatedmetagenomicdata" class="anchor"></a>Example in curatedMetagenomicData</h2>
<p>For this example, we use the VogtmannE 2016 dataset in cMD. The dataset has 52 cases of colorectal cancer and 58 controls.</p>
<p>First, download the data, extract the expressionset, and only take responses for which we have a value for study condition.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="no">vogtmann</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/curatedMetagenomicData/man/curatedMetagenomicData.html">curatedMetagenomicData</a></span>(<span class="st">"VogtmannE_2016.metaphlan_bugs_list.stool"</span>, <span class="kw">dryrun</span> <span class="kw">=</span> <span class="fl">FALSE</span>)</pre></body></html></div>
<pre><code>## Working on VogtmannE_2016.metaphlan_bugs_list.stool</code></pre>
<pre><code>## using temporary cache /tmp/Rtmp7MmAsT/BiocFileCache</code></pre>
<pre><code>## snapshotDate(): 2020-07-10</code></pre>
<pre><code>## see ?curatedMetagenomicData and browseVignettes('curatedMetagenomicData') for documentation</code></pre>
<pre><code>## downloading 1 resources</code></pre>
<pre><code>## retrieving 1 resource</code></pre>
<pre><code>## loading from cache</code></pre>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="no">vm.eset</span> <span class="kw">&lt;-</span><span class="no">vogtmann</span><span class="kw">[[</span><span class="fl">1</span>]]
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="no">vm.eset</span>$<span class="no">study_condition</span>, <span class="kw">useNA</span><span class="kw">=</span><span class="st">"ifany"</span>)</pre></body></html></div>
<pre><code>## 
## control     CRC    &lt;NA&gt; 
##      52      52       6</code></pre>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="no">vm.sub</span> <span class="kw">&lt;-</span> <span class="no">vm.eset</span>[,!<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">vm.eset</span>$<span class="no">study_condition</span>)]
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="no">vm.sub</span>$<span class="no">study_condition</span>, <span class="kw">useNA</span><span class="kw">=</span><span class="st">"ifany"</span>)</pre></body></html></div>
<pre><code>## 
## control     CRC 
##      52      52</code></pre>
<p>First we conduct a basic analysis to examine whether there are any statistically significant differences in taxa of microbiota by study condition. We use a forgiving alpha of 0.10 for adjusted p-values. We find 5 significant taxa in this crude analysis.</p>
<div class="sourceCode" id="cb17"><html><body><pre class="r"><span class="no">vm.pseq</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/curatedMetagenomicData/man/ExpressionSet2phyloseq.html">ExpressionSet2phyloseq</a></span>(<span class="no">vm.sub</span>)</pre></body></html></div>
<pre><code>## Warning: `data_frame()` is deprecated as of tibble 1.1.0.
## Please use `tibble()` instead.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_warnings()` to see where this warning was generated.</code></pre>
<div class="sourceCode" id="cb19"><html><body><pre class="r"><span class="no">dx_only</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/phyloseq_to_deseq2.html">phyloseq_to_deseq2</a></span>(<span class="no">vm.pseq</span>, ~ <span class="no">study_condition</span>)</pre></body></html></div>
<pre><code>## converting counts to integer mode</code></pre>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<div class="sourceCode" id="cb22"><html><body><pre class="r"><span class="no">dx_only</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span>(<span class="no">dx_only</span>, <span class="kw">test</span><span class="kw">=</span><span class="st">"Wald"</span>, <span class="kw">fitType</span><span class="kw">=</span><span class="st">"parametric"</span>)</pre></body></html></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>## -- replacing outliers and refitting for 58 genes
## -- DESeq argument 'minReplicatesForReplace' = 7 
## -- original counts are preserved in counts(dds)</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb32"><html><body><pre class="r"><span class="no">res</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span>(<span class="no">dx_only</span>, <span class="kw">cooksCutoff</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="no">alpha</span> <span class="kw">=</span> <span class="fl">0.10</span>
<span class="no">sigtab</span> <span class="kw">=</span> <span class="no">res</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">res</span>$<span class="no">padj</span> <span class="kw">&lt;</span> <span class="no">alpha</span>), ]
<span class="no">sigtab</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span>(<span class="fu">as</span>(<span class="no">sigtab</span>, <span class="st">"data.frame"</span>), <span class="fu">as</span>(<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span>(<span class="no">vm.pseq</span>)[<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">sigtab</span>), ], <span class="st">"matrix"</span>))
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">sigtab</span>)</pre></body></html></div>
<pre><code>##                                      baseMean log2FoldChange     lfcSE     stat
## g__Escherichia                       1.160319       3.131873 0.8680873 3.607785
## s__Ruminococcus_torques              2.057123       1.221872 0.3181531 3.840515
## s__Escherichia_coli                  1.021863       3.114437 0.8976217 3.469654
## t__Ruminococcus_torques_unclassified 1.317457       1.711670 0.4640449 3.688588
## t__Escherichia_coli_unclassified     1.021863       3.114437 0.8976217 3.469654
##                                            pvalue       padj  Kingdom
## g__Escherichia                       0.0003088219 0.05507325 Bacteria
## s__Ruminococcus_torques              0.0001227764 0.05507325 Bacteria
## s__Escherichia_coli                  0.0005211292 0.05576082 Bacteria
## t__Ruminococcus_torques_unclassified 0.0002255023 0.05507325 Bacteria
## t__Escherichia_coli_unclassified     0.0005211292 0.05576082 Bacteria
##                                              Phylum               Class
## g__Escherichia                       Proteobacteria Gammaproteobacteria
## s__Ruminococcus_torques                  Firmicutes          Clostridia
## s__Escherichia_coli                  Proteobacteria Gammaproteobacteria
## t__Ruminococcus_torques_unclassified     Firmicutes          Clostridia
## t__Escherichia_coli_unclassified     Proteobacteria Gammaproteobacteria
##                                                  Order             Family
## g__Escherichia                       Enterobacteriales Enterobacteriaceae
## s__Ruminococcus_torques                  Clostridiales    Lachnospiraceae
## s__Escherichia_coli                  Enterobacteriales Enterobacteriaceae
## t__Ruminococcus_torques_unclassified     Clostridiales    Lachnospiraceae
## t__Escherichia_coli_unclassified     Enterobacteriales Enterobacteriaceae
##                                            Genus              Species
## g__Escherichia                       Escherichia                 &lt;NA&gt;
## s__Ruminococcus_torques                  Blautia Ruminococcus_torques
## s__Escherichia_coli                  Escherichia     Escherichia_coli
## t__Ruminococcus_torques_unclassified     Blautia Ruminococcus_torques
## t__Escherichia_coli_unclassified     Escherichia     Escherichia_coli
##                                                                 Strain
## g__Escherichia                                                    &lt;NA&gt;
## s__Ruminococcus_torques                                           &lt;NA&gt;
## s__Escherichia_coli                                               &lt;NA&gt;
## t__Ruminococcus_torques_unclassified Ruminococcus_torques_unclassified
## t__Escherichia_coli_unclassified         Escherichia_coli_unclassified</code></pre>
<p>Next we adjust (deconfound) for categorized age. We find two significant results:</p>
<div class="sourceCode" id="cb34"><html><body><pre class="r"><span class="no">dx_age</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/phyloseq_to_deseq2.html">phyloseq_to_deseq2</a></span>(<span class="no">vm.pseq</span>, ~ <span class="no">study_condition</span> + <span class="no">age_category</span>)</pre></body></html></div>
<pre><code>## converting counts to integer mode</code></pre>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<div class="sourceCode" id="cb37"><html><body><pre class="r"><span class="no">dx_age</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span>(<span class="no">dx_age</span>, <span class="kw">test</span><span class="kw">=</span><span class="st">"Wald"</span>, <span class="kw">fitType</span><span class="kw">=</span><span class="st">"parametric"</span>)</pre></body></html></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>## -- replacing outliers and refitting for 60 genes
## -- DESeq argument 'minReplicatesForReplace' = 7 
## -- original counts are preserved in counts(dds)</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb47"><html><body><pre class="r"><span class="no">res</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span>(<span class="no">dx_age</span>, <span class="kw">cooksCutoff</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">contrast</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"study_condition"</span>, <span class="st">"CRC"</span>, <span class="st">"control"</span>))
<span class="no">alpha</span> <span class="kw">=</span> <span class="fl">0.10</span>
<span class="no">sigtab</span> <span class="kw">=</span> <span class="no">res</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">res</span>$<span class="no">padj</span> <span class="kw">&lt;</span> <span class="no">alpha</span>), ]
<span class="no">sigtab</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span>(<span class="fu">as</span>(<span class="no">sigtab</span>, <span class="st">"data.frame"</span>), <span class="fu">as</span>(<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span>(<span class="no">vm.pseq</span>)[<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">sigtab</span>), ], <span class="st">"matrix"</span>))
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">sigtab</span>)</pre></body></html></div>
<pre><code>##                                      baseMean log2FoldChange     lfcSE     stat
## s__Ruminococcus_torques              2.057123       1.225518 0.3199339 3.830534
## t__Ruminococcus_torques_unclassified 1.317457       1.706224 0.4661006 3.660634
##                                            pvalue       padj  Kingdom
## s__Ruminococcus_torques              0.0001278654 0.06730097 Bacteria
## t__Ruminococcus_torques_unclassified 0.0002515924 0.06730097 Bacteria
##                                          Phylum      Class         Order
## s__Ruminococcus_torques              Firmicutes Clostridia Clostridiales
## t__Ruminococcus_torques_unclassified Firmicutes Clostridia Clostridiales
##                                               Family   Genus
## s__Ruminococcus_torques              Lachnospiraceae Blautia
## t__Ruminococcus_torques_unclassified Lachnospiraceae Blautia
##                                                   Species
## s__Ruminococcus_torques              Ruminococcus_torques
## t__Ruminococcus_torques_unclassified Ruminococcus_torques
##                                                                 Strain
## s__Ruminococcus_torques                                           &lt;NA&gt;
## t__Ruminococcus_torques_unclassified Ruminococcus_torques_unclassified</code></pre>
<div class="sourceCode" id="cb49"><html><body><pre class="r"><span class="no">dx_age_gender</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/phyloseq_to_deseq2.html">phyloseq_to_deseq2</a></span>(<span class="no">vm.pseq</span>, ~ <span class="no">study_condition</span> + <span class="no">age_category</span> + <span class="no">gender</span>)</pre></body></html></div>
<pre><code>## converting counts to integer mode</code></pre>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<div class="sourceCode" id="cb52"><html><body><pre class="r"><span class="no">dx_age_gender</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span>(<span class="no">dx_age_gender</span>, <span class="kw">test</span><span class="kw">=</span><span class="st">"Wald"</span>, <span class="kw">fitType</span><span class="kw">=</span><span class="st">"parametric"</span>)</pre></body></html></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>## -- replacing outliers and refitting for 45 genes
## -- DESeq argument 'minReplicatesForReplace' = 7 
## -- original counts are preserved in counts(dds)</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb62"><html><body><pre class="r"><span class="no">res</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span>(<span class="no">dx_age_gender</span>, <span class="kw">cooksCutoff</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">contrast</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"study_condition"</span>, <span class="st">"CRC"</span>, <span class="st">"control"</span>))
<span class="no">alpha</span> <span class="kw">=</span> <span class="fl">0.10</span>
<span class="no">sigtab</span> <span class="kw">=</span> <span class="no">res</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">res</span>$<span class="no">padj</span> <span class="kw">&lt;</span> <span class="no">alpha</span>), ]
<span class="no">sigtab</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span>(<span class="fu">as</span>(<span class="no">sigtab</span>, <span class="st">"data.frame"</span>), <span class="fu">as</span>(<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span>(<span class="no">vm.pseq</span>)[<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">sigtab</span>), ], <span class="st">"matrix"</span>))
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">sigtab</span>)</pre></body></html></div>
<pre><code>##                                      baseMean log2FoldChange     lfcSE     stat
## s__Ruminococcus_torques              2.057123        1.22658 0.3202308 3.830298
## t__Ruminococcus_torques_unclassified 1.317457        1.73073 0.4655552 3.717562
##                                            pvalue       padj  Kingdom
## s__Ruminococcus_torques              0.0001279881 0.05380886 Bacteria
## t__Ruminococcus_torques_unclassified 0.0002011546 0.05380886 Bacteria
##                                          Phylum      Class         Order
## s__Ruminococcus_torques              Firmicutes Clostridia Clostridiales
## t__Ruminococcus_torques_unclassified Firmicutes Clostridia Clostridiales
##                                               Family   Genus
## s__Ruminococcus_torques              Lachnospiraceae Blautia
## t__Ruminococcus_torques_unclassified Lachnospiraceae Blautia
##                                                   Species
## s__Ruminococcus_torques              Ruminococcus_torques
## t__Ruminococcus_torques_unclassified Ruminococcus_torques
##                                                                 Strain
## s__Ruminococcus_torques                                           &lt;NA&gt;
## t__Ruminococcus_torques_unclassified Ruminococcus_torques_unclassified</code></pre>
<p>As mentioned in the confounding section, estimates obtained for the relationship between the exposure and the outcome have been adjusted for the included confounders (deconfounded). This means that the observed estimate is an averaged effect size across strata of the confounder(s).</p>
</div>
<div id="toy-problems" class="section level2">
<h2 class="hasAnchor">
<a href="#toy-problems" class="anchor"></a>Toy Problems</h2>
<ol style="list-style-type: decimal">
<li>Consider the DAG below. What variables should you control for if you want to measure the effect of X on Y?</li>
</ol>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
<ol start="2" style="list-style-type: decimal">
<li>Create your own DAG (of a causal relationship of interest to you) then identify what variables you would need to control for to measure the causal effect of interest.</li>
<li>One important subject that I did not touch on in this workshop was generalizability or transportability of causal findings (applying the results to an external population). What issues might arise when doing so? How can a DAG be used to resolve issues of generalizability and transportability?</li>
</ol>
</div>
<div id="conclusions" class="section level2">
<h2 class="hasAnchor">
<a href="#conclusions" class="anchor"></a>Conclusions</h2>
<p>Unfortunately there’s no silver bullet for confounding or bias in R or Bioconductor. Instead, we as researchers must carefully consider possible sources of bias–ideally as we begin designing the study. DAGs are helpful in providing a clear visualization of the hypothesized causal mechanism and identifying potential confounders.</p>
<p>Unmeasured confounding presents a particularly difficult problem for causal inference. Because the confounder was not measured, we cannot easily deconfound for it.</p>
<p>It is important for us to remember that the definition of a cause is rooted in the counterfactual: we need to find a way to come as close as possible to observing what would have happened if a participant’s exposure status had been different.</p>
</div>
<div id="further-reading" class="section level2">
<h2 class="hasAnchor">
<a href="#further-reading" class="anchor"></a>Further Reading</h2>
<ul>
<li>Pearl and Mackenzie <em>Book of Why: The New Science of Causal Effect</em> (easy summer beach read!)</li>
<li>Hernan and Robins <em>Causal Inference</em> available free online: <a href="https://www.hsph.harvard.edu/miguel-hernan/causal-inference-book/" class="uri">https://www.hsph.harvard.edu/miguel-hernan/causal-inference-book/</a>
</li>
<li>Pearl <em>Causality: Models, Reasoning, and Inference</em>
</li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Levi Waldron.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
